src/hosts.rs

```diff
@@ -0,0 +1,323 @@
+    OfflineSoft,
+    //when a server is put into OFFLINE_HARD mode, no new connections are created toward that server and the existing **free **connections are ** immediately dropped**, while backend connections currently associated with a client session are dropped as soon as the client tries to use them. This is equivalent to deleting the server from a hostgroup. Internally, setting a server in OFFLINE_HARD status is equivalent to deleting the server
+    OfflineHard,
+}
+
+impl fmt::Display for HostStatus {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            HostStatus::Online => write!(f, "ONLINE"),
+            HostStatus::Shunned => write!(f, "SHUNNED"),
+            HostStatus::OfflineSoft => write!(f, "OFFLINE_SOFT"),
+            HostStatus::OfflineHard => write!(f, "OFFLINE_HARD"),
+        }
+    }
+}
+
+impl From<String> for HostStatus {
+    fn from(s: String) -> Self {
+        match s.to_uppercase().as_str() {
+            "ONLINE" => HostStatus::Online,
+            "SHUNNED" => HostStatus::Shunned,
+            "OFFLINE_SOFT" => HostStatus::OfflineSoft,
+            "OFFLINE_HARD" => HostStatus::OfflineHard,
+            _ => HostStatus::Online,
+        }
+    }
+}
+
+/// Represents a Readyset host
+pub struct Host {
+    hostname: String,
+    port: u16,
+    status: HostStatus,
+    conn: Option<Conn>,
+}
+
+impl Host {
+    /// Creates a new `Host` instance with the given hostname and port.
+    /// The connection to the host is established during the creation of the instance.
+    /// If the connection fails, the `conn` field will be `None`.
+    /// If the connection is successful, the `conn` field will contain the connection.
+    ///
+    /// # Arguments
+    ///
+    /// * `hostname` - The hostname of the host.
+    /// * `port` - The port number of the host.
+    ///
+    /// # Returns
+    ///
+    /// A new `Host` instance.
+    pub fn new(hostname: String, port: u16, status: String, config: &Config) -> Host {
+        let conn = match Conn::new(
+            OptsBuilder::new()
+                .ip_or_hostname(Some(hostname.clone()))
+                .tcp_port(port)
+                .user(Some(config.readyset_user.clone()))
+                .pass(Some(config.readyset_password.clone()))
+                .prefer_socket(false),
+        ) {
+            Ok(conn) => conn,
+            Err(err) => {
+                eprintln!("Failed to establish connection: {}", err);
+                return Host {
+                    hostname,
+                    port,
+                    status: HostStatus::from(status),
+                    conn: None,
+                };
+            }
+        };
+
+        Host {
+            hostname,
+            port,
+            status: HostStatus::from(status),
+            conn: Some(conn),
+        }
+    }
+
+    /// Gets the hostname of the host.
+    ///
+    /// # Returns
+    ///
+    /// The hostname of the host.
```

Comment:
nit: This is a lot of vertical space for a comment for an accessor function

Reply:
I have used the same comment formatting for all functions. Do you want to remove the empty lines ?

Reply:
No, I don't think I'm asking for you to do that, and if you have a consistent thing going on, I don't care that much that it be changed.

I think the value of comments for accessor functions is pretty low, as this tells the reader the same thing (get the hostname) 3 times (including the name of the function).  If you can comfortably make these types of comments just 1 line, I do think that would be an improvement.

Feel free to ignore this advice.